src/portable/st/stm32_fsdev/dcd_stm32_fsdev.c

```diff
@@ -512,29 +493,42 @@ static void dcd_ep_ctr_tx_handler(uint32_t wIstr)
 
   // Verify the CTR_TX bit is set. This was in the ST Micro code,
   // but I'm not sure it's actually necessary?
-  if((wEPRegVal & USB_EP_CTR_TX) == 0U)
-  {
+  if ((wEPRegVal & USB_EP_CTR_TX) == 0U) {
     return;
   }
 
   /* clear int flag */
   pcd_clear_tx_ep_ctr(USB, EPindex);
 
-  xfer_ctl_t * xfer = xfer_ctl_ptr(ep_addr);
-  if((xfer->total_len != xfer->queued_len)) /* TX not complete */
+  xfer_ctl_t *xfer = xfer_ctl_ptr(ep_addr);
+
+  /* Ignore spurious int */
```

Comment:
do we still have issue with this spurious interrupt ? I wondered if it is caused by incorrect double buffer configuration previously.

Reply:
It's still necessary since Tx interrupt can't be masked, if I disable the EP the host will not happy seeing NAK.

Another way to fix is in classes driver's xfer_cb, always call usbd_edpt_xfer for iso even if there is nothing to send.

Reply:
No need to change the class driver, this is specifci to this dcd only. I see host send IN token while have nothing to send and ISO does not have NAK --> triggered interrupt with zero byte data. I think this is all good now. Though I think we should make it more ISO explicit thing (making changes atm)